tests/unit-tests/test-class-sensei-course-manual-enrolment-provider.php

```diff
@@ -0,0 +1,221 @@
+
+		$this->assertTrue( isset( $providers[ Sensei_Course_Manual_Enrolment_Provider::get_id() ] ), '`manual` provider key should be set' );
+		$this->assertTrue( $providers[ Sensei_Course_Manual_Enrolment_Provider::get_id() ] instanceof Sensei_Course_Manual_Enrolment_Provider, '`manual` provider should be of class Sensei_Course_Manual_Enrolment_Provider' );
+	}
+
+	/**
+	 * This provider should handle every course.
+	 */
+	public function testHandlesEnrolmentAlways() {
+		$course_id = $this->getSimpleCourseId();
+		$provider  = $this->getManualEnrolmentProvider();
+
+		$this->assertTrue( $provider->handles_enrolment( $course_id ), 'Manual provider should handle enrolment for all courses' );
+	}
+
+	/**
+	 * Tests to make sure legacy isn't migrated because they might have installed on Sensei 3.0.0 and then are now upgrading to 3.1.0.
+	 */
+	public function testIsEnrolledLegacyNoMigrateVersion() {
+		$provider   = $this->getManualEnrolmentProvider();
+		$course_id  = $this->getSimpleCourseId();
+		$student_id = $this->getStandardStudentUserId();
+
+		// Start course progress for the student and simulate a legacy enrolment.
+		$this->legacyEnrolStudentStartCourseProgress( $student_id, $course_id );
+
+		$this->assertFalse( get_option( 'sensei_enrolment_legacy' ), 'The site-wide legacy enrolment flag should not have been set.' );
+
+		$is_enrolled = $provider->is_enrolled( $student_id, $course_id );
+		$this->assertFalse( $this->wasLegacyEnrolmentChecked( $student_id, $course_id ), 'Legacy enrolment status for user should not have been checked.' );
+		$this->assertFalse( $is_enrolled, 'The user should have not been enrolled because they were not upgrading from a pre-3.0 version.' );
+	}
+
+	/**
+	 * Tests to make sure enrolment isn't given when the `sensei_is_legacy_enrolled` filters to a false value.
+	 */
+	public function testIsEnrolledLegacyNoMigrateFilter() {
+		$provider   = $this->getManualEnrolmentProvider();
+		$course_id  = $this->getSimpleCourseId();
+		$student_id = $this->getStandardStudentUserId();
+
+		// Start course progress for the student and simulate a legacy enrolment.
+		$this->legacyEnrolStudentStartCourseProgress( $student_id, $course_id );
+		$this->simulateUpgradingFromSensei2ToSensei3();
+		add_filter( 'sensei_is_legacy_enrolled', '__return_false' );
+
+		$this->assertNotEmpty( get_option( 'sensei_enrolment_legacy' ), 'The site-wide legacy enrolment flag should have been set.' );
+
+		$is_enrolled = $provider->is_enrolled( $student_id, $course_id );
+		$this->assertTrue( $this->wasLegacyEnrolmentChecked( $student_id, $course_id ), 'Legacy enrolment status for user should have been checked.' );
+		$this->assertFalse( $is_enrolled, 'The user should not have been enrolled due to legacy enrolment being blocked from the filter.' );
+	}
+
+	/**
+	 * Tests to make sure enrolment isn't provided on migration when the user doesn't have course progress.
+	 */
+	public function testIsEnrolledLegacyNoMigrateNoProgress() {
```

Comment:
I don't think this test does what its name and comment implies?

Reply:
Yup! Good catch. Looks like it was testing for proper enrollment, which happens below. Fixed in e3f26d3a5a914404f09b22ee5b68b5f581c8629f.